cdap-ui/app/main.js

```diff
@@ -178,15 +178,23 @@ angular
           if (
               $rootScope.currentUser && !myHelpers.objectQuery(config, 'data', 'profile_view')
              ) {
-            angular.extend(config, {
+
+            var extendConfig = {
               user: $rootScope.currentUser || null,
               headers: {
                 'Content-Type': 'application/json',
-                // Accessing stuff from $rootScope is bad. This is done as to resolve circular dependency.
-                // $http <- myAuthPromise <- myAuth <- $http <- $templateFactory <- $view <- $state
-                Authorization: ($rootScope.currentUser.token ? 'Bearer ' + $rootScope.currentUser.token: null)
               }
-            });
+            };
+
+            // This check is added because of HdInsight gateway security.
+            // If we set Authorization to null, it strips off their Auth token
```

Comment:
I thought we wanted to add everything to the same Authorization property (both theirs and ours) and let CDAP take what is valid for us. If this removes theirs Auth would it work in a secure environment?